### DISCUSSION TITLE
Any combined shortcut for turning wifi on + calibre wireless connect?

### DISCUSSION THREAD

**Discussion by bheeshmpita** [koreader/koreader#11801] [Q&A] [answered]
Currently i have set a shortcut for turning wifi on but i was eager to know does any shortcut exist that can achieve above mentioned fuctions?
Another query is while the device is connected to the calibre wirelessly if the user turns wifi off or the device gets to the sleep mode without ejecting the device first at calibre, can it corrupt the device storage or harm it in any other way?

**Comment by mergen3107** [comment]
The thing with multiple actions in one gesture profile (shortcut in this context) is that there is no guaranteed order of actions. Also, it might take various time until Wi-Fi network is picked up after you turn on Wi-Fi, and profiles don't have a "pause". 

So you might end up with mismatch in timing or just wrong order.

**Reply by mergen3107** [reply]
On the other hand, if you just make a profile gesture shortcut to just calibre sync, it should ask for Wi-Fi network in a dialogue (if this plugin was written well, that is :D)

**Comment by bheeshmpita** [comment]
Thanks, can you help me with the second part of the query too ?

**Reply by mergen3107** [reply]
For the second issue, I don't think it should corrupt files, unlike wired connection. 

I didn't test it though (I don't use calibre for wireless transfers at all)

**Comment by pazos** [comment, marked as answer]
next nightly will have calibre wireless start/stop actions, thanks to https://github.com/koreader/koreader/pull/11806

Use the start action to connect to calibre. 
Avoid calls to stop. Eject the device from calibre instead.

You won't need to care about wifi, it will be requested if something is missing.
